tests/e2e_logging.py

```diff
@@ -774,6 +774,35 @@ def test_historical_query(network, args):
     return network
 
 
+@reqs.description("Attempt to read historical state from missing files")
+@reqs.supports_methods("/app/log/private", "/app/log/private/historical")
+@reqs.no_http2()
+@app.scoped_txs()
+def test_historical_query_on_missing_files(network, args):
+
+    new_node = network.create_node("local://localhost")
+    network.join_node(
+        new_node,
+        args.package,
+        args,
+        copy_ledger=False,
+        target_node=network.find_any_backup(),
+        from_snapshot=True,
+    )
+    network.trust_node(new_node, args)
+
+    LOG.info("Check historical queries return ACCEPTED")
+    with new_node.client("user0") as c:
+        for seqno in [2, 50, 100]:
```

Comment:
This test won't fit nicely into the suites, because of these hardcoded seqnos and assumption they're in term 2.

I think we can use `/node/commit` + `/node/tx` to sample valid TxIDs, and ask for those. And while we're here we could confirm the node can still distinguish future, unknown TxIDs from known-but-in-a-hole.

(After dicsussion, this is deliberately _not_ in the suite, so not needed now)

Reply:
It won't indeed, and needs several refinements at least:

1. retire the node that's been added
2. determine the view dynamically
3. find out the snapshot txid, and ask for transactions in that range, rather than assume 100+

With that said, I think this basic test in e2e_logging is already reliable, and helped us spot the issue with forwarding.

Reply:
All done now.